File: Personal_Web_UI/src/app/dtos/Project.ts
export class GetProject {
    id?: number;
    projectName?: string;
    projectDescription?: string;
    projectUrl?: string;
    imageUrl?: string;
}

export class CreateProject {
    ProjectName?: string;
    ProjectDescription?: string;
    ProjectUrl?: string;
    ImageUrl?: string;

    constructor(projectName?: string, ProjectDescription?: string, projectUrl?: string, ImageUrl?: string) {
        this.ProjectName = projectName;
        this.ProjectDescription = ProjectDescription;
        this.ProjectUrl = projectUrl;
        this.ImageUrl = ImageUrl;
    }
}

export class UpdateProject {
    id?: number;
    projectName?: string;
    projectDescription?: string;
    projectUrl?: string;
    imageUrl?: string;

    constructor(id?: number, projectName?: string, ProjectDescription?: string, projectUrl?: string, imageUrl?: string){
        this.id = id;
        this.projectName = projectName;
        this.projectDescription = ProjectDescription;
        this.projectUrl = projectUrl;
        this.imageUrl = imageUrl;
    }
}

File: Personal_Web_API/Personal_Web_API/Dtos/Project.cs
namespace Personal_Web_API.Dtos
{
	public record GetProject (
		int Id,
		string ProjectName,
		string ProjectDescription,
		string ProjectUrl,
		string ImageUrl
	);

	public class CreateProject
	{
		public string ProjectName { get; set; }
		public string ProjectDescription { get; set; }
		public string ProjectUrl { get; set; }
		public string ImageUrl { get; set; }
		public int ProjectUserId { get; set; }
	}

	public record UpdateProject (
		int Id,
		string ProjectName,
		string ProjectDescription,
		string ProjectUrl,
		string ImageUrl
	);
}
